VEC.ASM

```diff
@@ -46,4 +46,35 @@ proc vec_dot
     ret
 endp vec_dot
 
+proc vec_mul_float
+    arg @@v:PTR Vec, @@f:dword 
+    local @@temp:dword
+    uses eax, ebx
+
+    mov eax, [@@v]
+
+    fld [eax+Vec.x]
+    fld [@@f]
```

Comment:
You don't have to push the [@@f], you can just add it to the fmul instruction so this could become
```suggestion
    fmul [@@f]
```